packages/mobile/src/consumerIncentives/ConsumerIncentivesHomeScreen.tsx

```diff
@@ -148,14 +223,25 @@ export default function ConsumerIncentivesHomeScreen() {
   const isSupercharging = userIsVerified && hasBalanceForSupercharge
   const tokenToSupercharge = useTokenToSupercharge()
 
-  const onPressCTA = () => {
-    if (userIsVerified) {
+  const { availableRewards, loading: loadingAvailableRewards } = useFetchAvailableRewards()
+  const claimRewardsLoading = useSelector((state) => state.supercharge.loading)
+  const canClaimRewards = availableRewards.length > 0
+  const dispatch = useDispatch()
+
+  const onPressCTA = async () => {
+    if (canClaimRewards) {
+      dispatch(claimRewards(availableRewards))
+    } else if (userIsVerified) {
       navigate(Screens.FiatExchangeOptions, { isCashIn: true })
     } else {
       navigate(Screens.VerificationEducationScreen, { hideOnboardingStep: true })
     }
     ValoraAnalytics.track(RewardsEvents.rewards_screen_cta_pressed, {
-      buttonPressed: userIsVerified ? RewardsScreenCta.CashIn : RewardsScreenCta.VerifyPhone,
+      buttonPressed: canClaimRewards
+        ? RewardsScreenCta.ClaimRewards
+        : userIsVerified
+        ? RewardsScreenCta.CashIn
+        : RewardsScreenCta.VerifyPhone,
```

Comment:
nit: at this point it would be more readable and less error prone for future changes to duplicate the calls in the previous if statements.